### PR TITLE
Cannode ESC control

### PR DESCRIPTION
**Pull request by dakejahl**
This PR adds support for controlling PWM ESCs from a cannode, which receives the actuator control output data from the host flight controller over UAVCAN.

I added a new module into the `px4_fmu-v4_cannode` build called `pwm_out_cannode`. This is a pared down version of `pwm_out`, which simply operates as a work item on a high priority work queue, running on publication of `actuator_outputs` (published from `uavcannode`), with the responsibilities of configuring and updating the PWM outputs.

This has been tested using a Pixhawk4 as the host, and a Pixracer as the cannode.
https://review.px4.io/plot_app?log=a6da6557-4933-4853-8f92-56d782ceaf44

I tested both 200hz and 400hz for the ESC update rate (**UAVCAN_ESC_RATE**), and saw no noticeable difference in flight performance.

### CODE REVIEW

**Comment by dakejahl** [comment]
FYI @JacobCrabill

**Comment by JacobCrabill** [comment]
Sweet, I'll take a look next week and give it a try

**Comment by JacobCrabill** [comment]
Tested this yesterday - works as expected.  Will be fantastic once we have a few options for getting RPM measurements for each ESC too (a topic for a later PR).

One thing I've noticed is that actuator_outputs on the cannode sometimes has random values in the "unused" entries (and the last entry is simply NaN); @dakejahl could this be cleaned up?  Maybe memset all actuators to 0 before filling in the used indices

File: src/drivers/uavcannode/UavcanNode.cpp [hunk 1]
```diff
@@ -330,9 +341,29 @@ UavcanNode::esc_raw_command_sub_cb(const uavcan::ReceivedDataStructure<uavcan::e
 		}
 	}
 
+	// Now remap the indices
+	remap_esc_indices(outputs);
```
Comment (JacobCrabill) [inline src/drivers/uavcannode/UavcanNode.cpp:345]: :+1:

**Comment by dakejahl** [comment]
Moved the branch here 
https://github.com/dakejahl/Firmware/tree/dev/uavcannode_esc

**Pull request closed without merging**